jekyll/_cci2/caching-strategy.md

```diff
@@ -0,0 +1,273 @@
+---
+layout: classic-docs
+title: "Caching strategies"
+description: "This document is a guide to the various caching strategies available for managing dependency caches in CircleCI."
+categories: [optimization]
+order: 50
+version:
+- Cloud
+- Server v3.x
+- Server v2.x
+---
+
+Caching is one of the most effective ways to make jobs faster on CircleCI. By reusing the data from previous jobs, you also reduce the cost of fetch operations. Caching is project-specific, and there are a number of strategies to help optimize caches for effectiveness and storage optimization.
+
+* TOC
+{:toc}
+
+## Cache optimization
+When setting up caches for your projects, the goal would be a 10x - 20x ROI (return on investment). The following tips can help you achieve this.
+
+### Use cache keys that are less strict
+### Avoid unnecessary workflow reruns
+### Split cache keys by directory
+### Combine jobs in a workflow if no affect on workflow length
+### Reorder jobs to create meaningful workflows
+### Check for language-specific caching tips
+### Check cache is actually being restored as well as saved
+
+
+## Partial dependency caching strategies
+{: #partial-dependency-caching-strategies }
+
+Some dependency managers do not properly handle installing on top of partially restored dependency trees.
+
+{% raw %}
+
+```yaml
+steps:
+  - restore_cache:
+      keys:
+        - gem-cache-{{ arch }}-{{ .Branch }}-{{ checksum "Gemfile.lock" }}
+        - gem-cache-{{ arch }}-{{ .Branch }}
+        - gem-cache
+```
+{% endraw %}
+
+In the above example, if a dependency tree is partially restored by the second or third cache keys, some dependency managers will incorrectly install on top of the outdated dependency tree.
+
+Instead of a cascading fallback, a more stable option is a single version-prefixed cache key:
+
+{% raw %}
+
+```yaml
+steps:
+  - restore_cache:
+      keys:
+        - v1-gem-cache-{{ arch }}-{{ .Branch }}-{{ checksum "Gemfile.lock" }}
+```
+
+{% endraw %}
+
+Since caches are immutable, this strategy allows you to regenerate all of your caches by incrementing the version. This is useful in the following scenarios:
+
+- When you change the version of a dependency manager like `npm`.
+- When you change the version of a language like Ruby.
+- When you add or remove dependencies from your project.
+
+The stability of partial dependency caching is dependent on your dependency manager. Below is a list of common dependency managers, recommended partial caching strategies, and associated justifications.
```

Comment:
"...dependency caching is dependent on your dependency manager." -- that's a lot of versions of "dependent" in one sentence. Maybe change it to something like "...dependency caching is reliant on your dependency manager.

Reply:
ewww yeah thanks for catching this!